lfsapi/endpoint_finder.go

```diff
@@ -26,14 +26,28 @@ const (
 	defaultRemote              = "origin"
 )
 
+type Access struct {
+	mode AccessMode
+	url  string
+}
+
+// Returns a copy of an AccessMode with the mode upgraded to newMode
+func (a *Access) Upgrade(newMode AccessMode) Access {
```

Comment:
Hmm. When we talked about this approach, I thought that there was more work to do in order to upgrade from one access mode to another.

I don't think it's bad at all thought that this function is short. In fact, I'd rather have a short function that introduces no mutable behavior than a short one that does (and I'd prefer that before a _longer_ function that introduces mutable behavior).